src/ATen/ExpandUtils.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "ATen/Tensor.h"
+#include <sstream>
+
+namespace at {
+
+std::tuple<Tensor> expand_inplace(const Tensor &tensor, const Tensor &to_expand) {
```

Comment:
Why do we use a tuple with a single element, instead of return a Tensor? Is it for the codegen below?

Reply:
Just to simplify the codegen.

Reply:
i.e. the 3-tensor-arg case is identical codegen wise (modulo the number of arguments, returns, which the script already handles), i.e.
```
std::tie(b_mask) = expand_inplace(self, mask);
std::tie(b_tensor1, b_tensor2) = expand_inplace(self, tensor1, tensor2);
```